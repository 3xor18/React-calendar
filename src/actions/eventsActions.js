import Swal from 'sweetalert2';
import { fetchConToken } from '../helpers/fecth';
import { prepareEvents } from '../helpers/prepareEvents';
import { types } from '../types/types';

const eventAddNewEvent = (event) => ({
	type: types.eventAddNew,
	payload: event,
});

export const eventSetActive = (event) => ({
	type: types.eventSetActive,
	payload: event,
});

export const eventClearActiveNote = () => ({
	type: types.eventClearActiveEvent,
});

//acciones
export const eventStartAddNew = (event) => {
	return async (dispatch, getState) => {
		console.log('event30:', event);
		const { uid, name } = getState().auth;
		try {
			const resp = await fetchConToken(
				'events',
				event,
				'POST'
			);
			const body = await resp.json();
			if (body.ok) {
				event.id = body.event.id;
				event.user = { _id: uid, name: name };
				console.log('event42:', event);
				dispatch(eventAddNewEvent(event));
			}
		} catch (error) {
			console.error(error);
		}
	};
};

export const eventStartLoading = () => {
	return async (dispatch) => {
		try {
			const resp = await fetchConToken('events');
			const body = await resp.json();

			const events = prepareEvents(body.eventos);

			dispatch(eventLoaded(events));
		} catch (error) {
			console.error(error);
		}
	};
};

const eventLoaded = (events) => ({
	type: types.eventLoaded,
	payload: events,
});

export const eventStartUpdate = (event) => {
	return async (dispatch) => {
		try {
			const resp = await fetchConToken(
				`events/${event.id}`,
				event,
				'PUT'
			);
			const body = await resp.json();

			if (body.ok) {
				dispatch(eventUpdated(event));
			} else {
				Swal.fire('Error', body.msg, 'error');
			}
		} catch (error) {
			console.error(error);
		}
	};
};

const eventUpdated = (event) => ({
	type: types.eventUpdated,
	payload: event,
});

export const eventStartDelete = () => {
	return async (dispatch, getState) => {
		try {
			const {
				id,
			} = getState().calendar.activeEvent;
			const resp = await fetchConToken(
				`events/${id}`,
				{},
				'DELETE'
			);
			const body = await resp.json();

			if (body.ok) {
				dispatch(eventDeleted());
			} else {
				Swal.fire('Error', body.msg, 'error');
			}
		} catch (error) {
			console.error(error);
		}
	};
};

const eventDeleted = () => ({
	type: types.eventDeleted,
});

export const eventLogOut = () => ({
	type: types.eventLogOut,
});
