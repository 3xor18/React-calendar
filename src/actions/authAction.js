import Swal from 'sweetalert2';
import {
	fetchConToken,
	fetchSinToken,
} from '../helpers/fecth';
import { types } from '../types/types';
import { eventLogOut } from './eventsActions';

export const startLogin = (email, password) => {
	return async (dispatch) => {
		const resp = await fetchSinToken(
			'auth',
			{
				email,
				password,
			},
			'POST'
		);
		const body = await resp.json();
		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem(
				'token-init-date',
				new Date().getTime()
			);
			dispatch(
				login({
					uid: body.user.uid,
					name: body.user.name,
				})
			);
		} else {
			console.log(body.error);
			let error;
			if (body.errors?.email)
				error = body.errors.email.msg;
			else error = body.error;
			Swal.fire('Error', error, 'error');
		}
	};
};

const login = (user) => {
	return {
		type: types.authLogin,
		payload: user,
	};
};

export const startRegister = (
	name,
	email,
	password
) => {
	return async (dispatch) => {
		const resp = await fetchSinToken(
			'auth/new',
			{
				name,
				email,
				password,
			},
			'POST'
		);
		const body = await resp.json();
		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem(
				'token-init-date',
				new Date().getTime()
			);
			console.log(body.user);
			dispatch(
				login({
					uid: body.user._id,
					name: body.user.name,
				})
			);
		} else {
			let error;
			if (body.errors?.email)
				error = body.errors.email.msg;
			if (body.errors?.password)
				error = body.errors.password.msg;
			else error = body.error;
			Swal.fire('Error', error, 'error');
		}
	};
};

export const startChecking = () => {
	return async (dispatch) => {
		const resp = await fetchConToken(
			'auth/renew',
			{}
		);
		const body = await resp.json();
		if (body.ok) {
			localStorage.setItem('token', body.token);
			localStorage.setItem(
				'token-init-date',
				new Date().getTime()
			);
			dispatch(
				login({
					uid: body.uid,
					name: body.name,
				})
			);
		} else {
			dispatch(checkingFinish());
		}
	};
};

const checkingFinish = () => ({
	type: types.authCheckingFinish,
});

export const starLogOut = () => {
	return (dispatch) => {
		localStorage.clear();
		dispatch(eventLogOut());
		dispatch(logOut());
	};
};

const logOut = () => ({
	type: types.authLogout,
});
