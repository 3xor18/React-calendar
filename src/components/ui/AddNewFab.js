import React from 'react';
import { useDispatch } from 'react-redux';
import { uiOpenModalAction } from '../../actions/uiAction';

export const AddNewFab = () => {
	const dispatch = useDispatch();
	const openModal = (e) => {
		dispatch(uiOpenModalAction());
	};
	return (
		<button
			className="btn btn-primary fab"
			onClick={openModal}
		>
			<i className="fas fa-plus"></i>
		</button>
	);
};
