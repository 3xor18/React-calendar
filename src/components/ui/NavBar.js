import React from 'react';
import {
	useDispatch,
	useSelector,
} from 'react-redux';
import { starLogOut } from '../../actions/authAction';

export const NavBar = () => {
	const dispatch = useDispatch();
	const { name } = useSelector(
		(state) => state.auth
	);
	const handleLogout = () => {
		dispatch(starLogOut());
	};
	return (
		<div className="navbar navbar-dark  bg-dark mb-4">
			<span className="navbar-brand">
				{name}!!
			</span>
			<button
				className="btn btn-outline-danger"
				onClick={handleLogout}
			>
				<i className="fas fa-sign-out-alt"></i>
				<span className=""> Salir</span>
			</button>
		</div>
	);
};
