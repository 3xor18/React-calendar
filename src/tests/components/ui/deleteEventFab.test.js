import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/eventsActions';
jest.mock(
	'../../../actions/eventsActions',
	() => ({
		eventStartDelete: jest.fn(),
	})
);

const middleWares = [thunk];
const mockStore = configureStore(middleWares);
const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<DeleteEventFab />
	</Provider>
);

describe('test en <DeleteEventFab />', () => {
	test('match con snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de llamar al eventStarDelete al hacer click', () => {
		wrapper.find('button').prop('onClick')();

		expect(eventStartDelete).toHaveBeenCalled();
	});
});
