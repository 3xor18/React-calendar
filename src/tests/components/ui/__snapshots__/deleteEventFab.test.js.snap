// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en <DeleteEventFab /> match con snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fa fa-trash"
  />
  <span
    className=""
  >
    Borrar Evento
  </span>
</button>
`;
